### DISCUSSION TITLE
No sound or sound dropping after a few minutes

### DISCUSSION THREAD

**Discussion by paulrouget** [aclap-dev/video-downloadhelper#6] [Help and Q&A] [answered]
Apparently, some users do not have sound in some cases.

- https://www.livestreamde.com/ard-live/
- https://www3.nhk.or.jp/news/movie.html

**Comment by paulrouget** [comment]
Also issues with sound being stopping after a few minutes.

**Comment by paulrouget** [comment]
Previously discussed here: https://github.com/aclap-dev/vdhcoapp/issues/197

**Comment by paulrouget** [comment]
Part of the issue has been identified. The upcoming v8.2 should improve the situation.

**Comment by paulrouget** [comment, marked as answer]
v8.2 announced here: https://github.com/aclap-dev/video-downloadhelper/discussions/13
With coapp here: https://www.downloadhelper.net/install-coapp-beta

Those should fix the sound issues.

**Comment by GrampaWildWilly** [comment]
Refer to this discussion:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/30eftZvT0SA

In there you will find the URL of a German language ASMR video on YouTube.  The primary complaint there is that the 4K webm gets a strange error.  Yes, that error is still occurring.  There has been nothing released yet that addresses that issue.  However, that ASMR video exhibited another anomalous behavior with the 1920x1080 mp4 variant.  The video was downloading fine but the audio was cutting off before the end of the video.  With this new beta CoApp, that problem is corrected.

Another YouTube video exhibiting the same short audio problem was this one:

https://www.youtube.com/watch?v=JCbHKQCkRpU

This one has no 4K variants to cloud the issue.  The 1920x1080 mp4 variant now downloads with a complete video track, which it has done all along, but it also now downloads a complete audio track, which is a correction of a previous fault.

So this CoApp beta is looking very good.

I must point out one bit of weirdness.  I'm getting this:

C:\DownloadHelper CoApp>dir
 Volume in drive C is Windows 11
 Volume Serial Number is 2066-6331

 Directory of C:\DownloadHelper CoApp

01/30/2024  08:15    `<`DIR`>`          .
01/26/2024  06:08        49,135,616 ffmpeg.exe
01/26/2024  06:08        49,002,496 ffprobe.exe
01/26/2024  06:08               468 google.json
01/26/2024  06:08               468 microsoft.json
01/26/2024  06:08               301 mozilla.json
01/30/2024  08:15            68,717 uninstall.exe
01/26/2024  06:08        38,884,945 vdhcoapp.exe
               7 File(s)    137,093,011 bytes
               1 Dir(s)  1,810,604,277,760 bytes free

C:\DownloadHelper CoApp>vdhcoapp.exe --version
2.0.14

C:\DownloadHelper CoApp>vdhcoapp.exe --info
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.14",
  "binary": "C:\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\WildWilly2",
  "converterBinary": "C:\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

It just says it's CoApp 2.0.14.  I would have hoped that it would have been 2.0.14 beta or 2.0.14a1 beta or even 2.0.15.  Something, anything to distinguish it from the 2.0.14 that we already had.  If there is something here that uniquely identifies this CoApp as the new beta, I don't see it.  When there's a beta of the VDH browser extension, that is reflected in its version number.  We should have the same sort of thing with the CoApp beta.

**Comment by GrampaWildWilly** [comment]
There should be a link somewhere on this page:

https://github.com/aclap-dev/vdhcoapp

It should link to a page where the CoApp betas are listed, sort of like the page on the VDH web site that lists all the beta versions of the VDH browser extension for Firefox.  There's a link on that page for all the regular CoApp releases but there should be a link right below that for the CoApp betas.

**Comment by paulrouget** [comment]
For the most recent (as in, beta version), use this link: https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-Installation

**Comment by GrampaWildWilly** [comment]
That's not at all obvious that it's the CoApp beta.  The word beta appears nowhere on that page.  The CoApp beta should be somewhere under https://github.com/aclap-dev/vdhcoapp.  There's a link on the right side of that page that says Releases.  If you click that, it takes you to the repository of regular CoApps.  You need to put another link right below that one that takes you to a repository of CoApp betas.  It is not logical to keep a repository of code under a Wiki, & it is not logical to keep a repository of CoApps under the VDH extension area.  We have an area here on GitHub for the CoApp.  That's where these betas need to live.  This may be fine for now but it really needs to be normalized.

Speaking of the word beta, the word beta appears nowhere in the stuff you get from --info & --version.  This can lead to confusion.

**Reply by paulrouget** [reply]
> That's not at all obvious that it's the CoApp beta.

It doesn't need to be.

> The CoApp beta should be somewhere under https://github.com/aclap-dev/vdhcoapp

No. Like it says there, that place is for developers. And Beta means nothing to developers (rolling releases).

> You need to put another link right below that one that takes you to a repository of CoApp betas.

No reasons either.

> It is not logical to keep a repository of code under a Wiki

There's no rules that says we can't.

> it is not logical to keep a repository of CoApps under the VDH extension area

VDH extension area?

> We have an area here on GitHub for the CoApp

Like I say in the CoApp repo, that's for developers only.

> That's where these betas need to live.

No.

> This may be fine for now but it really needs to be normalized.

You're making a ton of assumptions here :) I'm doing rolling releases.

**Comment by GrampaWildWilly** [comment]
Hmmm......  The CoApp area is only for developers.  Well then I'm just confused that a user-accessible repository of CoApp versions is in a developers-only area.

If a version is a beta, you need to use the word beta somewhere near there to label it as a beta.  It doesn't look like a beta.  It looks the same as the regular version.  Same concept for how you named the beta.  The file needs a distinct name, same as the VDH browser extension betas all get unique names that are distinct from the regular releases.  At the moment, this CoApp beta is called 2.0.14 & that's the same as the regular release.  Also, the --info & --version output doesn't say anything about a beta.  Is it or isn't it a beta?  I haven't checked but is this beta now residing on this page? https://github.com/aclap-dev/vdhcoapp/releases.  It doesn't look like it to me.  That page says 2.0.14 was released 5 days ago but this 2.0.14 beta is something we learned about from you upthread here just yesterday.  By giving the beta the exact same identification as the regular release, you have thoroughly confused me.  Maybe I'm alone in my confusion & everything is perfectly obvious to everybody else.

When I say VDH extension area I mean https://github.com/aclap-dev/video-downloadhelper.  Like https://github.com/aclap-dev/vdhcoapp is the CoApp area.  On this page https://github.com/aclap-dev both the VDH area & the CoApp area are labelled Public.  It was never apparent to me that the CoApp area was meant for developers only.

As for the rules of what should go in a Wiki, my understanding of a Wiki is that it's for documentation.  Logic would tell me to expect code to be somewhere other than mixed in with documentation.  I know we can improvise whatever we please here.  But it just confuses me even more to have code mixed into a documentation area.  It just doesn't make sense to me.

**Comment by paulrouget** [comment]
> Hmmm...... The CoApp area is only for developers. Well then I'm just confused that a user-accessible repository of CoApp versions is in a developers-only area.

Because it's an Open Source project.

> If a version is a beta, you need to use the word beta somewhere near there to label it as a beta. It doesn't look like a beta. It looks the same as the regular version. Same concept for how you named the beta. The file needs a distinct name, same as the VDH browser extension betas all get unique names that are distinct from the regular releases. At the moment, this CoApp beta is called 2.0.14 & that's the same as the regular release. Also, the --info & --version output doesn't say anything about a beta. Is it or isn't it a beta? I haven't checked but is this beta now residing on this page? https://github.com/aclap-dev/vdhcoapp/releases. It doesn't look like it to me. That page says 2.0.14 was released 5 days ago but this 2.0.14 beta is something we learned about from you upthread here just yesterday. By giving the beta the exact same identification as the regular release, you have thoroughly confused me. Maybe I'm alone in my confusion & everything is perfectly obvious to everybody else.

See it like this: the coapp has no channels (stable / beta / dev). It's a rolling release. The coapp beta link is the coapp version required for vdh beta.
 
> When I say VDH extension area I mean https://github.com/aclap-dev/video-downloadhelper. Like https://github.com/aclap-dev/vdhcoapp is the CoApp area. On this page https://github.com/aclap-dev both the VDH area & the CoApp area are labelled Public. It was never apparent to me that the CoApp area was meant for developers only.

Well now you know :)

> As for the rules of what should go in a Wiki, my understanding of a Wiki is that it's for documentation. Logic would tell me to expect code to be somewhere other than mixed in with documentation. I know we can improvise whatever we please here. But it just confuses me even more to have code mixed into a documentation area. It just doesn't make sense to me.

I don't know why you're talking about code. There's no code in the wiki.

**Reply by GrampaWildWilly** [reply]
Right.  I meant executable code, not source code.  Also, when I said file name, I meant the name of the file I downloaded to get the beta of the CoApp.  Of course the CoApp has to be named vdhcoapp.exe.  But if I hadn't been paying attention, I would have accidentally replaced the installer file for the regular CoApp 2.0.14 with the one for the beta because the files have the same name.  I like to keep at least the current version of software plus one older one in case I need to go back to the older version.

**Reply by paulrouget** [reply]
These are just links to the latest relevant coapp. And they are named `latest` because I want to abstract the versioning.

I understand you want to archive the different versions of the addon and the coapp for your personal use, but for the sake of simplicity I'd rather keep the filename version less.

Otherwise it requires a mechanic to rename the files and website and that's work I'm trying to avoid for now.
